### PR TITLE
Conditional requests for GitHub version checks #132

### PR DESCRIPTION
**Pull request by silentsokolov**
Reason lack of test: difficult to moke the go-github library

### CODE REVIEW

File: updates/updates.go [hunk 1]
```diff
@@ -2,49 +2,34 @@ package updates
...
 	"github.com/hashicorp/go-version"
 	"github.com/pkg/errors"
 	"github.com/yext/edward/common"
 )
 
 // UpdateAvailable determines if a newer version is available given a repo
 func UpdateAvailable(owner, repo, currentVersion, cachePath string, logger common.Logger) (bool, string, error) {
-	printf(logger, "Checking for cached version at %v", cachePath)
-	isCached, latestVersion, err := getCachedVersion(cachePath)
+	dickCache := diskcache.New(cachePath)
```
Comment (theothertomelliott) [inline updates/updates.go]: spelling: diskCache

Comment (theothertomelliott) [inline updates/updates.go]: spelling: diskCache

File: cmd/root.go [hunk 2]
```diff
@@ -167,7 +167,7 @@ func initConfig() {
 
 func checkUpdateAvailable(checkUpdateChan chan interface{}) {
 	defer close(checkUpdateChan)
-	updateAvailable, latestVersion, err := updates.UpdateAvailable("yext", "edward", common.EdwardVersion, filepath.Join(home.EdwardConfig.Dir, ".updatecache"), logger)
+	updateAvailable, latestVersion, err := updates.UpdateAvailable("yext", "edward", common.EdwardVersion, filepath.Join(home.EdwardConfig.Dir, ".cache"), logger)
```
Comment (theothertomelliott) [inline cmd/root.go]: This seems a little generic, especially if we ever decide to add other caching.
Perhaps .cache/version would be a better option?

Comment (silentsokolov) [reply cmd/root.go]: `diskcache` expects directory (himself creates a file) aka `.cache/5872f43ir84yfgc3fg484f`

Comment (silentsokolov) [reply cmd/root.go]: Agree, fix.

**Comment by theothertomelliott** [comment]
Thanks for making this change!

**Pull request merged**